src/app/news/news.component.ts

```diff
@@ -44,137 +38,131 @@ export class NewsListComponent implements OnInit, OnDestroy {
-    // Go to top of page after clicking to a different page.
-    window.scrollTo(0, 0);
-    this.loading = true;
-
-    this.tableParams.sortBy = '-dateAdded';
-    this.tableParams = this.tableTemplateUtils.updateTableParams(this.tableParams, pageNumber, this.tableParams.sortBy);
-    this.tableParams.sortBy = '-dateAdded';
-
-    this.searchService.getSearchResults(
-      this.tableParams.keywords,
-      'RecentActivity',
-      null,
-      pageNumber,
-      this.tableParams.pageSize,
-      this.tableParams.sortBy,
-      {},
-      true,
-      this.tableParams.sortBy
-    )
-      .takeUntil(this.ngUnsubscribe)
-      .subscribe((res: any) => {
-        if (res[0].data) {
-          this.tableParams.totalListItems = res[0].data.meta[0].searchResultsTotal;
-          this.recentActivities = res[0].data.searchResults;
-          this.tableTemplateUtils.updateUrl(this.tableParams.sortBy, this.tableParams.currentPage, this.tableParams.pageSize, null, this.tableParams.keywords);
-          this.setRowData();
-          this.loading = false;
-          this._changeDetectionRef.detectChanges();
-        } else {
-          alert('Uh-oh, couldn\'t load topics');
-          // activity not found --> navigate back to search
-          this.router.navigate(['/']);
-        }
-      });
+  onPageNumUpdate(pageNumber) {
+    this.tableData.currentPage = pageNumber;
+    this.submit();
   }
 
-  public onSubmit() {
-    // dismiss any open snackbar
-    // if (this.snackBarRef) { this.snackBarRef.dismiss(); }
-
-    // NOTE: Angular Router doesn't reload page on same URL
-    // REF: https://stackoverflow.com/questions/40983055/how-to-reload-the-current-route-with-the-angular-2-router
-    // WORKAROUND: add timestamp to force URL to be different than last time
-
-    const params = this.terms.getParams();
-    params['ms'] = new Date().getMilliseconds();
-    params['dataset'] = this.terms.dataset;
-    params['currentPage'] = this.tableParams.currentPage = 1;
-    params['sortBy'] = this.tableParams.sortBy = '-dateAdded';
-    params['keywords'] = this.tableParams.keywords;
-    params['pageSize'] = this.tableParams.pageSize = 10;
-
-    console.log('params =', params);
-    console.log('nav:', ['news', params]);
-    this.router.navigate(['news', params]);
+  onPageSizeUpdate(pageSize: IPageSizePickerOption) {
+    this.tableData.pageSize = pageSize.value;
+    if (this.tableData.pageSize === this.tableData.totalListItems) {
+      this.loading = true;
+    }
+    this.tableData.currentPage = 1;
+    this.submit();
+  }
+
+  async submit() {
+    delete this.queryParams.sortBy;
+    delete this.queryParams.currentPage;
+    delete this.queryParams.pageNumber;
+    delete this.queryParams.pageSize;
+
+    const params = { ...this.queryParams, ...this.tableTemplateUtils.getNavParamsObj(this.tableData) }
+
+    this.location.replaceState(
+      this.router.serializeUrl(
+        this.router.createUrlTree(
+          ['news'],
```

Comment:
Missing a leading `/` here?  This is appending an extra `news` to the URL.

Reply:
1342489